modules/launcher/src/main/java/org/mule/runtime/module/launcher/artifact/ArtifactMuleContextBuilder.java

```diff
@@ -0,0 +1,244 @@
+     * @param parentContext {@code MuleContext} that is parent of the one to be created.
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setParentContext(MuleContext parentContext)
+    {
+        this.parentContext = parentContext;
+        return this;
+    }
+
+    /**
+     * The artifact properties define key value pairs that can be referenced from within the configuration
+     * files.
+     *
+     * @param artifactProperties properties use for the artifact configuration
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setArtifactProperties(Map<String, String> artifactProperties)
+    {
+        checkArgument(artifactProperties != null, MULE_CONTEXT_ARTIFACT_PROPERTIES_CANNOT_BE_NULL);
+        this.artifactProperties = artifactProperties;
+        return this;
+    }
+
+    /**
+     * Sets a meaningful name to identify the artifact. If not provided a UUID will be used.
+     *
+     * @param artifactName name to use to identify the artifact.
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setArtifactName(String artifactName)
+    {
+        this.artifactName = artifactName;
+        return this;
+    }
+
+    /**
+     * Allows to set a listener that will be notified when the {@code MuleContext} is created,
+     * initialized or configured.
+     *
+     * @param muleContextListener listener of {@code MuleContext} notifications.
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setMuleContextListener(MuleContextListener muleContextListener)
+    {
+        this.muleContextListener = muleContextListener;
+        return this;
+    }
+
+    /**
+     * Sets the file location where the artifact is installed. Must be a directory.
+     *
+     * @param location directory where the artifact is installed
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setArtifactInstallationDirectory(File location)
+    {
+        checkArgument(location.isDirectory(), INSTALLATION_DIRECTORY_MUST_BE_A_DIRECTORY);
+        this.artifactInstallationDirectory = location;
+        return this;
+    }
+
+    /**
+     * Sets the classloader that must be used to execute all {@code MuleContext} tasks such as
+     * running flows, doing connection retries, etc.
+     *
+     * @param classloader classloader to use for executing logic within the {@code MuleContext}
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setExecutionClassloader(ClassLoader classloader)
+    {
+        this.executionClassLoader = classloader;
+        return this;
+    }
+
+    /**
+     * Sets the default encoding for the {@code MuleContext} if the use did not define one
+     * explicitly within the configuration.
+     *
+     * @param defaultEncoding default encoding to use within the {@code MuleContext}
+     * @return the builder
+     */
+    public ArtifactMuleContextBuilder setDefaultEncoding(String defaultEncoding)
```

Comment:
Does it not make sense to use Configuration object which currently holds encoding.  Or what is the relationship between this builder and MuleConfiguration?